app/code/Magento/InventoryAdminUi/Plugin/InventoryAdminUi/Ui/DataProvider/PreventDisablingDefaultSourcePlugin.php

```diff
@@ -0,0 +1,79 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\InventoryAdminUi\Plugin\InventoryAdminUi\Ui\DataProvider;
```

Comment:
Need to move plugin and declaration to `InventoryCatalogAdminUi` because
Default Source is concept of `InventoryCatalog`